Client/Frontend/Home/HomepageViewController.swift

```diff
@@ -269,7 +269,12 @@ class HomepageViewController: UIViewController, HomePanel, FeatureFlaggable {
 
         // Force the entire collectionview to re-layout
         viewModel.refreshData(for: traitCollection)
+        collectionView.reloadData()
         collectionView.collectionViewLayout.invalidateLayout()
+
+        // This pushes a reload to the end of the main queue after all the work assocated with
+        // rotating has been completed. This is important because some of the cells layout
+        // based on the screen state
```

Comment:
nit: I have a feeling we're missing a word here?
```suggestion
        // rotating has been completed. This is important because some of the cells layout are
        // based on the screen state
```